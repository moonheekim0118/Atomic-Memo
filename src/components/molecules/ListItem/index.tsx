import React from 'react';
import Span from '../../atoms/Span';
import MemoTitle from '../MemoTitle';
import styled from 'styled-components';
import { Link, useLocation } from 'react-router-dom';

// 네비게이션 바에 들어갈 메모 아이템 링크
// 메모 종류
// 마지막으로 완성한 시간
// 메모 제목

interface Props {
  title: string;
  time: string;
  kind: string;
  path: string;
}

const ListItem = ({ title, time, kind, path }: Props) => {
  const location = useLocation();

  return (
    <StyledListItem to={path} visiting={location.pathname === path}>
      <MemoTitle title={title} kind={kind} />
      <Span size="small" padding="2em 1em">
        {time}
      </Span>
    </StyledListItem>
  );
};

const StyledListItem = styled<{ visiting: boolean }>(Link)`
  display: flex;
  justify-content: space-between;
  text-decoration: none;

  width: 100%;
  height: 90px;
  text-align: center;
  background-color: ${(props) => (props.visiting ? '#f4f4f4' : '#fff')};

  transition: background-color 0.5s ease;
  &:hover {
    background-color: #f4f4f4;
  }
`;

export default ListItem;
