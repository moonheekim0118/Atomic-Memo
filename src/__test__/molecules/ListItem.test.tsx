import 'jsdom-global/register';
import React from 'react';
import ListItem from '../../components/molecules/ListItem';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';

describe('<ListItem/>', () => {
  let container = null;

  it('renders correctly', () => {
    container = mount(
      <MemoryRouter>
        <ListItem title="test" time="10" kind="Work" path="" />
      </MemoryRouter>
    );
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
