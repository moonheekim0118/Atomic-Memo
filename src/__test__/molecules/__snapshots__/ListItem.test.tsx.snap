// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem/> matches snpashot 1`] = `"<a class=\\"ListItem__StyledListItem-sc-1i7a48o-0 dUQKlR\\" href=\\"/\\"><div class=\\"MemoTitle__StyledMemoTitle-e33wpr-0 cPfMRA\\"><span width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 dZpPqp small\\"></span><span width=\\"5.5em\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 bvWVQT small\\">test</span></div><span width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 iExBFZ small\\">10</span></a>"`;
